bazel/repository_locations.bzl

```diff
@@ -987,6 +987,51 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         license = "Apache-2.0",
         license_url = "https://github.com/grpc-ecosystem/proto-field-extraction/blob/{version}/LICENSE",
     ),
+    com_google_ortools = dict(
+        project_name = "or-tools",
+        project_desc = "Software suite for combinatorial optimization.",
+        project_url = "https://github.com/google/or-tools",
+        version = "2384738a951561bf905f3435651310841c309b4c",
+        sha256 = "b374f96f4509ab8c6c185bb497f4284c3b309bcaf87d1478eb4ddda009d1f12d",
+        strip_prefix = "or-tools-{version}",
+        urls = ["https://github.com/google/or-tools/archive/{version}.zip"],
+        use_category = ["dataplane_ext"],
+        extensions = ["envoy.filters.http.grpc_json_transcoder", "envoy.filters.http.grpc_field_extraction"],
+        release_date = "2024-06-10",
+        cpe = "N/A",
+        license = "Apache-2.0",
+        license_url = "https://github.com/google/or-tools/blob/stable/LICENSE",
+    ),
```

Comment:
Scorecard score - [4.8](https://scorecard.dev/viewer/?uri=github.com/google/or-tools)

Reply:
What's the purpose of combinatorial optimization for the extension? I'd be suspicious of using any algorithms here since they are typically super-linear.

Reply:
/wait-any

or-tools is research software, can you explain what it's used for?

Reply:
Hi Kuat!

`or-tools` isn't directly used in the logging libraries added for the PML filter. However, it is required for the Proto Processing Libraries. [Reference](https://github.com/grpc-ecosystem/proto_processing_lib/blob/09e6cf5da15c8b9d32c6706b2517890200110bc8/proto_processing_lib/proto_scrubber/BUILD.bazel#L94C11-L94C29).

The library is using `ortools/base/map_util.h`.

In envoy, the build fails if I do not explicitly import the `or-tools` library. 
```
$ bazel build //source/extensions/filters/http/proto_message_logging/logging_util:logging_util
ERROR: /usr/local/google/home/dchakarwarti/.cache/bazel/_bazel_dchakarwarti/0a1ddbe361fc4585fd11ce5863817141/external/com_google_protoprocessinglib/proto_processing_lib/proto_scrubber/BUILD.bazel:73:11: no such package '@com_google_ortools//ortools/base': The repository '@com_google_ortools' could not be resolved: Repository '@com_google_ortools' is not defined and referenced by '@com_google_protoprocessinglib//proto_processing_lib/proto_scrubber:field_mask_node'
ERROR: Analysis of target '//source/extensions/filters/http/proto_message_logging/logging_util:logging_util' failed; build aborted: 
INFO: Elapsed time: 0.865s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (25 packages loaded, 840 targets configured)
```

Reply:
I think it was added because it's coming from Google internal gtl, and or-tools exported some fork of it at some point?  Can you re-work your dependency not to depend on a repo to use gtl helper - you can simply copy paste those two helpers directly into your code.

Reply:
A side question for these three proto libraries - are you somehow hooked into the internals of protobuf? Is there some hidden friend access? We can't update protobufs nearly as fast in OSS, so it'd be challenging if you use private proto interfaces.

Reply:
>  you can simply copy paste those two helpers directly into your code.

Sure, I removed the `ortools` dependency from here.

Additionally, I also removed `@com_google_proto_field_extraction` external repo dependency as it was already being used as `@com_google_protofieldextraction`.

Now, the only newly introduced dependency is of `@com_google_protoprocessinglib`.

>  are you somehow hooked into the internals of protobuf?

Actually, no. There is no hidden friend access. 

To add more context, this library is part of [grpc-ecosystem](https://github.com/grpc-ecosystem/proto_processing_lib) repo from which we have been using the field extraction and proto converter libraries as well.